plutus-contract/src/Plutus/Contract/Request.hs

```diff
@@ -151,6 +150,36 @@ waitNSlots n = do
   c <- currentSlot
   awaitSlot $ c + fromIntegral n
 
+-- | Wait until the given time.
+awaitTime ::
+    forall w s e.
+    ( AsContractError e
+    )
+    => POSIXTime
+    -> Contract w s e POSIXTime
+awaitTime s = pabReq (AwaitTimeReq s) E._AwaitTimeResp
+
+-- | Get the current time
+currentTime ::
+    forall w s e.
+    ( AsContractError e
+    )
+    => Contract w s e POSIXTime
+currentTime = pabReq CurrentTimeReq E._CurrentTimeResp
+
+-- | Wait for a number of seconds to pass.
+--
+-- Note: Currently, if n < length of a slot, then 'waitNSeconds' has no effect.
+waitNSeconds ::
+  forall w s e.
+  ( AsContractError e
+  )
+  => Natural
+  -> Contract w s e POSIXTime
+waitNSeconds n = do
+  t <- currentTime
+  awaitTime $ t + fromIntegral n
```

Comment:
I was about to say "but isn't the resolution picoseconds?" but no, it's seconds. So two things:
- I'm slightly confused because I thought we were doing time as unix picoseconds like `Data.Time` does, although I don't think it matters much.
- IMO a `Num` instance is a bit suspect since it lets you do things like this that are very easy to get wrong. Could we just provide `fromSeconds` instead?

Reply:
Ahh ok then. I'll change seconds to picoseconds.

Reply:
I didn't mean it was wrong! I actually don't remember whether we discussed it. @j-mueller do you remember?

Reply:
It doesn't make sense to have a resolution smaller than 1 slot length and the slot length is currently one second. I'm not sure what the chances are of it ever becoming smaller than a second. But maybe we should go with milliseconds to be on the safe side.

(We did discuss this but I couldn't find evidence of it on slack - only this somewhat related thread: https://input-output-rnd.slack.com/archives/C21UF2WVC/p1619443414258000)

Reply:
Alright, I see 2 options:

1. convert every to use milliseconds instead of seconds. `DiffSeconds` becomes, `DiffMilliSeconds`, `fromSeconds` becomes `fromMilliSeconds`, `waitNSeconds` becomes `waitNMilliSeconds`, etc.
2. keep functions like `waitNSeconds`, but allow user to use floating numbers. Ex. `waitNSeconds 0.010` which waits for 10 milliseconds. Therefore, `POSIXTime` won't use an Integer anymore, but a Double (although I don't know if Double is supported by PLC)

Reply:
Or: be happy with seconds and leave it as it is :D At any rate, I think let's do it in another PR.

I do think `fromIntegral` assuming the unit _is_ risky, though.

Reply:
Haha, alright. I'll do it in another PR.